django_emarsys/checks.py

```diff
@@ -0,0 +1,134 @@
+
+    Minimal test for this:
+
+    >>> from django.conf import settings
+    >>> from django.test.utils import override_settings
+    >>> @override_settings()
+    >>> def test_strange_dir_issue():
+    >>>     delattr(settings, 'DEBUG')
+    >>>     assert 'DEBUG' in dir(settings)
+    >>>     assert not hasattr(settings, 'DEBUG')
+    >>>
+    >>> test_strange_dir_issue()
+
+    """
+    return validate_configuration({key: getattr(settings, key)
+                                  for key in dir(settings)
+                                  if hasattr(settings, key)})
+
+
+def validate_configuration(config):
+    messages = []
+    if not config.get('EMARSYS_ACCOUNT'):
+        add_message(messages, Critical("EMARSYS_ACCOUNT not set"))
+
+    if not config.get('EMARSYS_PASSWORD'):
+        add_message(messages, Critical("EMARSYS_PASSWORD not set"))
+
+    if not config.get('EMARSYS_BASE_URI'):
+        add_message(messages, Critical("EMARSYS_BASE_URI not set"))
+
+    events = config.get('EMARSYS_EVENTS')
+    if not events:
+        events = {}
+        add_message(messages, Critical("EMARSYS_EVENTS not set"))
+
+    for event, event_params in events.items():
+        if not event:
+            add_message(
+                messages,
+                Warning("invalid event name: '{event}'".format(event=event)))
+
+        cache = {'name': set(), 'arg': set()}
+        for data in event_params:
+            for key in ('name', 'arg', 'model'):
+                validate_event_param(event, data, key, cache, messages)
+
+            for key in set(data.keys()) - {'name', 'arg', 'model'}:
+                add_message(
+                    messages,
+                    Warning("unexpected parameter key for event "
+                            "'{event}': '{key}'"
+                            .format(event=event, key=key)))
+
+    return messages
+
+
+def validate_event_param(event, data, key, cache, messages):
+    VALID_VALUES = {
+        'name': re.compile(r'^[a-zA-Z0-9_ ]+$'),
+        'arg': re.compile(r'^[a-z][a-z0-9_]*$'),
+        'model': re.compile(r'^[a-z][a-z0-9_]*[.][A-Z][a-zA-Z0-9_]*$'),
+    }
+
+    if key not in data:
+        add_message(messages,
+                    Error("missing parameter key for event '{event}': '{key}'"
+                          .format(event=event, key=key)))
+        return
+
+    value = data[key]
+    if key in cache:
+        if value in cache[key]:
+            Message = Error if key == 'arg' else Warning
+            add_message(
+                messages,
+                Message("reused parameter {key} for event '{event}': '{value}'"
+                        .format(key=key, value=value, event=event)))
+        else:
+            cache[key].add(value)
+
+    if value is None or not VALID_VALUES[key].match(value):
+        add_message(
+            messages,
+            Error("invalid parameter {key} for event '{event}': '{value}'"
+                  .format(key=key, value=value, event=event)))
```

Comment:
Yeah, I thought about this, but there is no "id" or anything. Worst case is that the key is the only one present.
It might be useful to output the whole `dict`.